### ISSUE TITLE
Importing single-spa-layout into new CRA build fails

### ISSUE THREAD

**Issue by alanquigley-toast** [single-spa/single-spa-layout#90]
In a default fresh Create React App install

```
    "single-spa": "^5.8.0",
    "single-spa-layout": "^1.1.3",
```

I get this error immediately after importing

```
./node_modules/single-spa-layout/dist/esm/single-spa-layout.min.js
Attempted import error: 'single-spa' does not contain a default export (imported as 'e').
```

Should this work?

**Comment by joeldenning** [comment]
This error indicates that you might be importing single-spa-layout into your code. Could you share the code that is importing single-spa-layout?

```js
// This won't work, since single-spa-layout doesn't export default
import singleSpaLayout from 'single-spa-layout';

// This will work
import * as singleSpaLayout from 'single-spa-layout';
```

**Comment by alanquigley-toast** [comment]
```js
import { registerApplication, start } from 'single-spa'
import {
  constructApplications,
  constructRoutes,
  constructLayoutEngine
} from 'single-spa-layout'
```

**Comment by alanquigley-toast** [comment]
```js
import * as singleSpaLayout from 'single-spa-layout'
```
Same error with none default

**Comment by joeldenning** [comment]
This is likely related to the workaround I put into place in https://github.com/single-spa/single-spa-layout/blob/1ad8e3cd02154eaa318b6a43bdda4386570878bf/src/isomorphic/constructRoutes.js#L15-L21

**Comment by alanquigley-toast** [comment]
Is there a previous version you'd like me to test?

**Comment by joeldenning** [comment]
We do not recommend using create-react-app for your single-spa root config, and single-spa-layout only should be installed in your root config. The root config should not use a UI framework, and you can use the single-spa CLI to generate a root config repo, including webpack configuration. [Root config explanation](https://single-spa.js.org/docs/faq/#should-i-have-a-parentroot-app-and-children-apps)

That said, I was able to reproduce the issue you reported and have put together the following fix: https://github.com/single-spa/single-spa-layout/pull/91

**Comment by alanquigley-toast** [comment]
Thank you 🙏

**Comment by joeldenning** [comment]
Released in https://github.com/single-spa/single-spa-layout/releases/tag/v1.1.5

**Comment by celian-garcia** [comment]
Hello, did the 1.1.5 fixed the issue ? 
I mean I am trying single-spa-layout in my project 
```json
  "dependencies": {
    "single-spa": "5.8.1",
    "single-spa-layout": "^1.1.5"
  },
```
```typescript
import { registerApplication, start } from 'single-spa';
import { constructApplications, constructLayoutEngine, constructRoutes } from 'single-spa-layout';
```

And I am facing a similar issue 
```
ERROR in ./node_modules/single-spa-layout/dist/esm/single-spa-layout.min.js 2:2995-3021
export 'default' (imported as 'e') was not found in 'single-spa' (possible exports: BOOTSTRAPPING, LOADING_SOURCE_CODE, LOAD_ERROR, MOUNTED, MOUNTING, NOT_BOOTSTRAPPED, NOT_LOADED, NOT_MOUNTED, SKIP_BECAUSE_BROKEN, UNMOUN
TING, UPDATING, addErrorHandler, checkActivityFunctions, ensureJQuerySupport, getAppNames, getAppStatus, getMountedApps, mountRootParcel, navigateToUrl, pathToActiveWhen, registerApplication, removeErrorHandler, setBootst
rapMaxTime, setMountMaxTime, setUnloadMaxTime, setUnmountMaxTime, start, triggerAppChange, unloadApplication, unregisterApplication)
 @ ./src/main.ts 2:0-98 14:17-32 15:23-44 23:23-44
```

**Comment by joeldenning** [comment]
When I put together the fix previously, I tested it with a normal webpack project, not in CRA. Since CRA uses webpack, I figured that it would work. I generally don't use CRA since its webpack config is hefty and difficult to modify. Additionally, we don't encourage using CRA as the single-spa root config, since single-spa-layout is frameworkless and root configs should not have a framework ([explanation](https://single-spa.js.org/docs/faq/#should-i-have-a-parentroot-app-and-children-apps)). Single-spa-layout should only be used in a single-spa root config.

All of that said, single-spa-layout should be importable into a CRA project for those who wish to disregard our advice 😄. I tried it just now and was able to reproduce the problem. Reopening this.
